test/infrastructure/docker/main.go

```diff
@@ -163,12 +164,20 @@ func setupChecks(mgr ctrl.Manager) {
 }
 
 func setupReconcilers(ctx context.Context, mgr ctrl.Manager) {
+	// Set our runtime client into the context for later use
+	runtimeClient, err := container.NewDockerClient()
+	if err != nil {
+		setupLog.Error(err, "unable to establish container runtime connection", "controller", "reconciler")
+		os.Exit(1)
+	}
+	ctx = runtimeClient.RuntimeInto(ctx)
+
 	if err := (&controllers.DockerMachineReconciler{
 		Client: mgr.GetClient(),
```

Comment:
Sorry should have written a bit more :)

This doesn't work entirely because the per-Reconcile context is a different one that the one we pass in here.

I wanted to suggest to:
* create the client in main.go (as you did in l.167-l.172)
* then store the containerClient like we do it with the CR client in the reconciler
* then inject that client at the top of the Reconcile funcs into the per-Reconcile context (`ctx = r.ContainerClient.RuntimeInto(ctx)`)

```suggestion
		Client: mgr.GetClient(),
		ContainerClient: containerClient
```